bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -82,16 +89,115 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                        bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                        int overrideMode = getOverrideMode();
+                        switch (overrideMode) {
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_SCHEDULE_NEXT:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        getNextSetPointChange(schedule));
+                                break;
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_TIME_ADD:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        LocalDateTime.now().plusMinutes(getOverrideTime()));
+                                break;
+                            default:
+                                bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private DailySchedules getDailySchedules() {
+        return this.getEvohomeBridge().getZoneSchedule("temperatureZone", this.getId());
+    }
+
+    private LocalDateTime getNextSetPointChange() {
+        return getNextSetPointChange(this.getDailySchedules());
+    }
+
+    private LocalDateTime getNextSetPointChange(DailySchedules schedule) {
+        if (schedule != null) {
+            int currentWeekDay = LocalDateTime.now().getDayOfWeek().getValue() - 1;
+            LocalDateTime nextSetPointChange = null;
+            for (DailySchedule daily : schedule.getSchedules()) {
+                // handle wrap around
+                int dayDelta = 0;
+                if (daily.getWeekday() < currentWeekDay) {
+                    // the schedule day is from earlier in the week so add 7 to get to the same day next week
+                    dayDelta = daily.getWeekday() + 7 - currentWeekDay;
+                } else {
+                    dayDelta = daily.getWeekday() - currentWeekDay;
+                }
+                LocalDate setPointDay = LocalDate.now().plusDays(dayDelta);
+                for (Switchpoint sp : daily.getSwitchpoints()) {
+                    LocalDateTime spNextOccurance = LocalDateTime.of(setPointDay,
+                            LocalTime.parse(sp.getTimeOfDay().toString()));
+                    if (spNextOccurance.isAfter(LocalDateTime.now())) {
+                        // setPoint is after now we are interested in it.
+                        if (nextSetPointChange == null || nextSetPointChange.isAfter(spNextOccurance)) {
+                            nextSetPointChange = spNextOccurance;
+                        }
                     }
                 }
             }
+            if (nextSetPointChange == null) {
+                return LocalDateTime.now().plusMinutes(getOverrideTime());
```

Comment:
Same as when `schedule` is null. If you initialize the default at the beginning of the method with this you can reduce three return to one. This would aid in readability as well.